code/datums/components/food/edible.dm

```diff
@@ -453,3 +453,15 @@ Behavior that's still missing from this component that original food items had t
 /datum/component/edible/proc/onCrossed(datum/source, mob/user)
 	SIGNAL_HANDLER
 	SEND_SIGNAL(parent, COMSIG_FOOD_CROSSED, user, bitecount)
+
+///Response to customization, inherit tastes and foodtypes if possible
+/datum/component/edible/proc/on_customized(datum/source, obj/item/I)
+	SIGNAL_HANDLER
+
+	var/datum/component/edible/E = I.GetComponent(/datum/component/edible)
```

Comment:
ditto

Reply:
Your suggestion of a signal on attack won't work, because then the attacker component won't have the parent atom's edible component if it exists. And getcomponent will still have to be called.

Reply:
Thats why you need to make a **new** signal for when food attacks something, as this is the concept of "When an edible component is used to attack something" event. The signal would automatically pass the edible component as that would be the sender.

Reply:
So then how does the parent's edible component find out about this? A second signal?

Reply:
Are you sure the sender is the edible component and not the item being attacked with?

Reply:
If you make a _new_ signal sent from the edible component when an edible component is used to attack, then yes, it would be sent by an edible component.